File: server/HomeWorkServices/HomeWorkServices/Models/Project.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace HomeWorkServices.Models
{
    public class Project : BaseEntity<int>
    {
        
        public Project()
        {
            ProjectTasks = new List<ProjectTask>();
        }
       
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string Priority { get; set; }
        public bool Completed { get; set; } = false;
        public ICollection<ProjectTask> ProjectTasks { get; set; }

    }
}

File: client/src/app/shared/models/Project.ts
import { ProjectTask } from './ProjectTask';

export class Project
{
    id:number;
    name:string;
    description:string;
    startDate?:Date;
    endDate?:Date;
    priority: 'low' | 'medium' | 'high';
    completed:boolean;
    spentHours:number;
    projectTasks:ProjectTask[]
}